src/main/kotlin/com/mineinabyss/idofront/serialization/recipes/SmokingRecipeIngredients.kt

```diff
@@ -0,0 +1,26 @@
+package com.mineinabyss.idofront.serialization.recipes
+
+import com.mineinabyss.idofront.serialization.SerializableItemStack
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import org.bukkit.NamespacedKey
+import org.bukkit.inventory.ItemStack
+import org.bukkit.inventory.Recipe
+import org.bukkit.inventory.RecipeChoice
+import org.bukkit.inventory.SmokingRecipe
+
+@Serializable
+@SerialName("smoking")
+class SmokingRecipeIngredients(
+    val input: SerializableItemStack,
+    val experience: Float,
+    val cookingTime: Int
+) : SerializableRecipeIngredients() {
+    override fun toRecipe(key: NamespacedKey, result: ItemStack, group: String): Recipe {
+        val recipe = SmokingRecipe(key, result, RecipeChoice.ExactChoice(input.toItemStack()), experience, cookingTime)
+
+        recipe.group = group
```

Comment:
Dont like this slightly reused code, perhaps you could remove the group parameter from this function and make another one with the group parameter which would run toRecipe then set the group

Reply:
- We spoke about this.

Bukkit bad and has no common interface apart from Recipe which doesn't have what we need